src/dotnet/commands/dotnet-run3/Run3Command.cs

```diff
@@ -87,30 +71,30 @@ private ITaskItem GetRunInformation()
 
             ProjectInstance projectInstance = new ProjectInstance(Project, globalProperties, null);
 
-            BuildRequestData buildRequestData = new BuildRequestData(projectInstance, new string[] { GetRunInformationTaskName });
-            BuildParameters buildParameters = new BuildParameters();
-
-            BuildResult result = BuildManager.DefaultBuildManager.Build(buildParameters, buildRequestData);
-
-            TargetResult runInfoResult;
-            if (!result.ResultsByTarget.TryGetValue(GetRunInformationTaskName, out runInfoResult))
+            string runProgram = projectInstance.GetPropertyValue("RunCommand");
+            if (string.IsNullOrEmpty(runProgram))
             {
-                throw new InvalidOperationException($"Could not find a target named '{GetRunInformationTaskName}' in your project. Please ensure 'dotnet run' supports this project.");
-            }
+                string outputType = projectInstance.GetPropertyValue("OutputType");
 
-            if (runInfoResult.ResultCode != TargetResultCode.Success)
-            {
-                throw new InvalidOperationException($"Could not get the run information for project {Project}. An internal MSBuild error has occured" + Environment.NewLine +
-                    runInfoResult.Exception?.ToString());
+                throw new GracefulException(string.Join(Environment.NewLine,
+                    "Unable to run your project.",
+                    "Please ensure you have a runnable project type and ensure 'dotnet run' supports this project.",
+                    $"The current OutputType is '{outputType}'."));
             }
 
-            ITaskItem runInfoItem = runInfoResult.Items.FirstOrDefault(i => i.ItemSpec == projectInstance.FullPath);
-            if (runInfoItem == null)
+            string runArguments = projectInstance.GetPropertyValue("RunArguments");
+            string runWorkingDirectory = projectInstance.GetPropertyValue("RunWorkingDirectory");
```

Comment:
If these aren't set, it works as expected:

1. no args are passed into the command.
2. The "working directory" will be empty string - which means use the current working directory.  See https://msdn.microsoft.com/en-us/library/system.diagnostics.processstartinfo.workingdirectory(v=vs.110).aspx